pkg/conv/conversions.go

```diff
@@ -82,3 +83,21 @@ func ToEnvVar(input map[string]string) []v1.EnvVar {
 	}
 	return out
 }
+
+// ToUlimits takes a slice of strings formatted in a particular way. It returns a new slice
+// with the strings parsed as Ulimits such that they can be used by docker. If any of the ulimit
+// strings are improperly formatted, they will be ignored.
+// Each element in the input should be formatted in one of the following ways:
+// nofile=512:1024
+// nofile=1024
+// cpu=2:4
+// cpu=6
+func ToUlimits(input []string) (out []*units.Ulimit, err error) {
+	for _, s := range input {
+		parsed, err := units.ParseUlimit(s)
+		if err == nil {
+			out = append(out, parsed)
+		}
+	}
+	return
```

Comment:
Naked returns are **dangerous**. This is not doing what you think it's doing. The assignment above (`parsed, err := units.ParseUlimit(s)`) is creating a new local `err` variable, not assigning to the return variable. Therefore, the returned error is always nil.

Also we should short-circuit if we detect an error, no reason why to continue processing the rest of the input.